src/cookbooks/bigquery/optimization.md

```diff
@@ -1,28 +1,165 @@
-    - Applying a `LIMIT` clause to a `SELECT *` query might not affect the amount of data read, depending on the table structure.
-      - Many of our tables are configured to use _clustering_ in which case a `LIMIT` clause does effectively limit the amount of data that needs to be scanned.
-      - Tables that include a `sample_id` field will usually have that as one of the clustering fields and you can efficiently scan random samples of users by specifying `WHERE sample_id = 0` (1% sample), `WHERE sample_id < 10` (10% sample), etc. This can be especially helpful with `main_summary`, `clients_daily`, and `clients_last_seen` which are very large tables and are all clustered on `sample_id`.
-      - To check whether your `LIMIT` and `WHERE` clauses are actually improving performance, you should see a lower value reported for actual "Data Scanned" by a query compared to the prediction ("This query will process X bytes") in STMO or the BigQuery UI.
-    - If you are experimenting with data or exploring data, use one of the [data preview options](https://cloud.google.com/bigquery/docs/best-practices-costs#preview-data) instead of `SELECT *`.
-        - Preview support is coming soon to BigQuery data sources in [re:dash](https://sql.telemetry.mozilla.org/)
-- Limit the amount of data scanned by using a date partition filter
-    - Tables that are larger than 1 TB will require that you provide a date partition filter as part of the query.
-    - You will receive an error if you attempt to query a table that requires a partition filter.
-        - `Cannot query over table 'moz-fx-data-shared-prod.telemetry_derived.main_summary_v4' without a filter over column(s) 'submission_date' that can be used for partition elimination`
-    - See [_Writing Queries_](./querying.md#writing-queries) for examples.
-- Reduce data before using a JOIN
-    - Trim the data as early in the query as possible, before the query performs a JOIN. If you reduce data early in the processing cycle, shuffling and other complex operations only execute on the data that you need.
-    - Use sub queries with filters or intermediate tables or views as a way of decreasing sides of a join, prior to the join itself.
+# Optimizing BigQuery Queries
+
+When writing a query using [STMO](https://sql.telemetry.mozilla.org) or the BigQuery console, you can improve performance and reduce costs by learning how data is stored, how databases function, and what you can change about a query to take advantage of the storage structure and the data function.
+
+[Queries are charged by data scanned at \$5 per terabyte (TB)](https://cloud.google.com/bigquery/pricing#on_demand_pricing) so each 200 gigabytes of data scanned will cost \$1: on tables with hundreds of TBs of data (like the [main ping table](../../datasets/pings.md#main-ping) or [`clients_daily`](../../datasets/batch_view/clients_daily/reference.md)), costs **can add up very quickly**.
```

Comment:
I don't want people to necessarily conflate reducing costs with improving performance. They are related but not the same. Some of the optimizations below will decrease costs, some won't. We could consider marking everything below that decreases costs, and note that here; or re-order it such that those are first.

Reply:
Good point, I guess the point to emphasize is that reducing costs entails reducing the amount of data scanned. I can take a stab at adjusting this text to send that message. We can make further changes later if desired.

Reply:
Just updated the article further, adding some text to the introduction and doing the breakdown you suggested. I think this is better! Lmk what you think (when you have time)

Reply:
I would change the following from:
When writing a query
To
When you write a query